rlmolecule/alphazero/alphazero_vertex.py

```diff
@@ -0,0 +1,30 @@
+import logging
+from abc import abstractmethod
+from typing import (
+    List,
+    Optional, Dict,
+)
+
+import numpy as np
+
+from rlmolecule.mcts.mcts_vertex import MCTSVertex
+from rlmolecule.tree_search.graph_search_state import GraphSearchState
+
+logger = logging.getLogger(__name__)
+
+
+class AlphaZeroVertex(MCTSVertex):
+    """
+    A class which implements the AlphaZero search methodology, with the assistance of a supplied
+    AlphaZeroGame implementation ("game").
+
+    Users must implement a `policy` function, that takes as inputs the next possible actions and returns a value
+    score for the current vertex and prior score for each child.
+    """
+
+    def __init__(self, state: GraphSearchState) -> None:
+        super().__init__(state)
+
+        self.child_priors: Optional[Dict['AlphaZeroVertex', float]] = None  # lazily initialized
+        self.policy_inputs: Optional[Dict[str, np.ndarray]] = None  # lazily initialized
```

Comment:
With the way we defined the `policy_predictions` / `get_value_estimate` function, any sort of `policy_inputs` would be internal to a derived class that actually had a policy model. These would likely need to get stored, etc., but I think we can't assume that their structure is a dictionary of numpy arrays.